### ISSUE TITLE
FOUC after Nuxt 3.10.2 update

### ISSUE THREAD

**Issue by mitjans** [nuxt/nuxt#25806]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v20.11.0
- Nuxt Version:     3.10.1
- CLI Version:      3.10.1
- Nitro Version:    -
- Package Manager:  pnpm@8.15.2
- Builder:          -
- User Config:      devtools, css
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

I haven't been able to reproduce it in Stackblitz, but the following steps should work:

1. Create a new project
    ```bash
    $ pnpx nuxi@latest init fouc
    ```

2. Create the following file `assets/css/main.css`:
    ```css
    body {
      background-color: greenyellow;
    }
    ```

3. Change `app.vue`:

    ```vue
    <template>
      <div>
        Hello world
      </div>
    </template>
    ```

4. Add the file to `nuxt.config.ts`:

    ```typescript
    export default defineNuxtConfig({
      css: ['assets/css/main.css'],
    })
    ```

5. Start the dev server and refresh the page multiple times:
    ```bash
    $ pnpm dev
    ```

You will notice that there's FOUC in the body styles. It is much more prominent when there are multiple styles applied, or a tool like Tailwindcss is used.
If you fix the Nuxt version to `3.10.1` with the following change in `package.json` it works correctly:

```json
"resolutions": {
  "nuxt": "3.10.1"
}
```

### Describe the bug

FOUC for global css styles passed to `nuxt.config.ts`.

### Additional context

I haven't been able to pinpoint the commit that causes this issue, but looking at the release notes of Nuxt 3.10.2 it might be related to the following PR:

- Inline entry styles before component styles (https://github.com/nuxt/nuxt/pull/25749)

**Comment by becem-gharbi** [comment]
I've noticed that this occurs on Windows and not Linux, with external and not inline styles, in dev and not in prod. Maybe this is related to Vite.

**Comment by danielroe** [comment]
Yes, this is a vite issue. It will be resolved with https://github.com/vitejs/vite/pull/15919

**Comment by patak-dev** [comment]
This should be fixed in vite@5.1.3. Thanks for the help debugging this regression @danielroe! ❤️

**Comment by becem-gharbi** [comment]
>This should be fixed in vite@5.1.3.

Unfortunately, the issue still exists 🤔

**Comment by mitjans** [comment]
@rylanharper you can pin a dependency using ["overrides"](https://pnpm.io/package_json#pnpmoverrides) or ["resolutions"](https://pnpm.io/package_json#resolutions) in your package.json file. Something like this should work while waiting for the next Nuxt release:

```
"resolutions": {
  "vite": "5.1.3"
}
```
